src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/RiskService.kt

```diff
@@ -86,7 +86,9 @@ internal class RiskService(
   suspend fun getAssessmentStatus(crn: String): String {
     val assessmentsResponse = fetchAssessments(crn)
     val latestAssessment =
-      assessmentsResponse.assessments?.sortedBy { LocalDateTime.parse(it.initiationDate).toLocalDate() }?.reversed()
+      assessmentsResponse.assessments
+        ?.filter { !it.initiationDate.isNullOrEmpty() }
+        ?.sortedBy { LocalDateTime.parse(it.initiationDate).toLocalDate() }?.reversed()
```

Comment:
We can sort on just the string and get rid of parsing entirely.  ISO formatted dates can be sorted as they are, that is one of the reasons why they are formatted the way they are - year, month, date.  They are indeed sorted lexigraphically, as you mentioned in your summary, but this gives exactly the same result had they been parsed and sorted chronologically as dates.
The more complication we have, the more chance of a bug (e.g. the one this PR is aiming to fix) so let's do the simplest thing that works.

Reply:
We shouldn't filter out the assessments with null initiation dates.  They will be the latest assessments if we don't have non-null assessments.  What we have to watch for is sort order of null/empty.  It might be that the best idea is to sort on { initiationDate ?: "1970-01-01" }, then we are clear that we are treating a null date as an old date and reverse will behave transparently and correctly.
Although on the subject of reversed - why don't we skip the reverse and just take last instead of first?

Reply:
Done 😌